components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -589,396 +958,394 @@ setup_boundary_exchange () {
-          const int ilev = idx % NumVerticalLevels;
+      auto phys = uview_nd<ScalarT,2> (reinterpret_cast<ScalarT*>(phys_ptrs(i).get()),
+                                       dim_p[0],
+                                       dim_p[1]);
+
+      if (is_state_field_dev(i)) {
+        auto dyn = uview_nd<ScalarT,5> (reinterpret_cast<ScalarT*>(dyn_ptrs(i).get()),
+                                        dim_d[0],
+                                        dim_d[1],
+                                        dim_d[2],
+                                        dim_d[3],
+                                        dim_d[4]);
+
+        const auto tr = Kokkos::TeamThreadRange(team, num_cols*dim_d[4]);
+        const auto f = [&] (const int idx) {
+          const int icol = idx/dim_d[4];
+          const int ilev = idx%dim_d[4];
 
           const auto& elgp = Kokkos::subview(lid2elgp,p2d(icol),Kokkos::ALL());
-          dyn(elgp[0],itl,elgp[1],elgp[2],ilev) = phys(icol,ilev);
-        });
+          dyn(elgp[0],time_levels(0).first,elgp[1],elgp[2],ilev) = phys(icol,ilev);
+        };
+        Kokkos::parallel_for(tr, f);
       } else {
-        auto dyn  = dyn_field.template get_reshaped_view<ScalarT****>();
-        const int NumVerticalLevels = dyn.extent_int(3);
-        Kokkos::parallel_for(RangePolicy(0,num_cols*NumVerticalLevels),
-                             KOKKOS_LAMBDA(const int idx) {
-          const int icol = idx / NumVerticalLevels;
-          const int ilev = idx % NumVerticalLevels;
+        auto dyn = uview_nd<ScalarT,4> (reinterpret_cast<ScalarT*>(dyn_ptrs(i).get()),
+                                        dim_d[0],
+                                        dim_d[1],
+                                        dim_d[2],
+                                        dim_d[3]);
+
+        const auto tr = Kokkos::TeamThreadRange(team, num_cols*dim_d[3]);
+        const auto f = [&] (const int idx) {
+          const int icol = idx/dim_d[3];
+          const int ilev = idx%dim_d[3];
 
           const auto& elgp = Kokkos::subview(lid2elgp,p2d(icol),Kokkos::ALL());
           dyn(elgp[0],elgp[1],elgp[2],ilev) = phys(icol,ilev);
-        });
+        };
+        Kokkos::parallel_for(tr, f);
       }
       break;
     }
-    case LayoutType::Vector3D:
+    case etoi(LayoutType::Vector3D):
     {
-      auto phys = phys_field.template get_reshaped_view<ScalarT***>();
-      const int dim = phys_dims[1];
-      if (itl>=0) {
-        auto dyn  = dyn_field.template get_reshaped_view<ScalarT******>();
-        const int NumVerticalLevels = dyn.extent_int(5);
-        Kokkos::parallel_for(RangePolicy(0,num_cols*dim*NumVerticalLevels),
-                             KOKKOS_LAMBDA(const int idx) {
-          const int icol =  idx / (dim*NumVerticalLevels);
-          const int idim = (idx / NumVerticalLevels) % dim;
-          const int ilev =  idx % NumVerticalLevels;
+      auto phys = uview_nd<ScalarT,3> (reinterpret_cast<ScalarT*>(phys_ptrs(i).get()),
```

Comment:
Not for this PR, but maybe when you do the team policy rework: you could make this even cleaner by having a routine

```
template<typename T, int N>
uview_nd<T,n> reshape (const Pointer& p, conost Dim& dim_specs) {
  // check inputs, maybe, and then reintepret case and create uview
}
```
and then in the code simply do
```
auto dyn = reshape<ScalarT,6>(dyn_ptr(i),dyn_dims(i));
```
It's mostly aesthetic, but might be easier to parse.